File: cs/src/tqs/RealTime.cs
//using Newtonsoft.Json;
using System.Text.Json;

namespace TQuant.Stralet
{

    public class RealTime
    {
        public delegate Stralet CreateStralet();

        public class Config
        {
            public string dapi_addr;
            public string tapi_addr;
        }

        static public void Run(Config cfg, CreateStralet creatae_stralet)
        {
            //var json = JsonConvert.SerializeObject(cfg);
            var json = JsonSerializer.Serialize(cfg);

            StraletWrap wrap = null;
            Impl.TqsDll.StraletCreator my_creatae_stralet = () =>
            {
                // FIXME
                var stralet = creatae_stralet();
                wrap = new StraletWrap(stralet);
                return wrap.handle;
            };

            Impl.TqsDll.tqs_rt_run(json, my_creatae_stralet);

            wrap = null;
        }
    }
}

File: java/src/main/java/com/acqusta/tquant/stralet/RealTime.java
package com.acqusta.tquant.stralet;

import com.acqusta.tquant.stralet.impl.StraletJni;
import com.fasterxml.jackson.databind.ObjectMapper;

public class RealTime {

    static public class Config {
        public String dapi_addr;
        public String tapi_addr;
    }

    static public void run(Config cfg, StraletCreator straletCreator) throws Exception {
        ObjectMapper mapper = new ObjectMapper();
        String txt = mapper.writeValueAsString(cfg);
        StraletJni.runRealTime(txt, straletCreator);
    }
}
